Initial implementation

javihernandez opened the pull request:
This is an initial implementation of this work. 
Full description is here: https://issues.gpii.net/browse/GPII-1575

jobara commented on .jshintrc:
```diff
@@ -0,0 +1,31 @@
+{
```
Seems like we might be making a shift to ESHint. Can you check in with @amb26 to see if this should be switched over.

amb26 replied:
Yes, our experiment has been successful. Please take linting config from this file: https://github.com/fluid-project/infusion/blob/master/.eslintrc.json and surrounding package.json and Gruntfile.js

jobara commented on README.md:
```diff
@@ -1,2 +1,39 @@
+
+The GPII for Google Chrome is an extension that allows Google Chrome to be configured by the GPII and improve the accessibility of the browser in different ways.
+
+Although the extension is on development right now, it includes support for:
```
Typo, should be "in development"

jobara commented on README.md:
```diff
@@ -1,2 +1,39 @@
+
+Follow these steps if you want to use the unpacked version of the extension:
+
+1. Visit *chrome://extensions* in your browser (or open up the Chrome menu by clicking the icon to the far right of the Omnibox:  The menu's icon is three horizontal bars. and select *Extensions* under the *Tools* menu to get to the same place).
```
Typos - I think it should be

Visit *chrome://extensions* in your browser. Alternatively, open the Chrome menu by clicking the icon to the far right of the Omnibus; the menu's icon is three horizontal bars.  Select *Extensions* under the *Tools* menu to open Chrome's extension settings.

jobara commented on extension/src/background.js:
```diff
@@ -0,0 +1,6 @@
+var gpii = fluid.registerNamespace("gpii");
+var chromeSettings = gpii.chrome.settings();
+
+console.log("Instantiated chromeSettings: " + chromeSettings);
```
I image we'll want to remove the console log before merging the pull request.

jobara commented on extension/src/lib/chromeSettings.js:
```diff
@@ -0,0 +1,80 @@
+        chromeVox: {
+            type: "gpii.chrome.extensionHolder",
+            options: {
+                extensionId: "kgejglhpjiefppelpmljglcjbhoiplfn",
```
Is this ID specific to a particular instance of the chromeVox extension?

javihernandez replied:
Yes,

it's ChromeVox' id, see:  https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en

jobara commented on extension/src/lib/chromeSettings.js:
```diff
@@ -0,0 +1,80 @@
+        fluid.each(settings, function (value, key) {
+            that.applier.change(["currentSettings", key], value);
+        });
+    }
```
I'm not sure I understand the need to null out `oldSettings` when `settings` is undefined. Also I wonder if you can get all of this behaviour done with using the [`modelRelay` system](http://docs.fluidproject.org/infusion/development/ModelRelay.html). This would also allow you to change the `onSettingsChange` invoker to use the [declarative style for triggering a change](http://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#declarative-style-for-triggering-a-change).

javihernandez replied:
> This would also allow you to change the onSettingsChange invoker to use the declarative style for triggering a change.

I've reworked this a bit, but how can I make use of the declarative style for triggering a change since I need to check whether settings are undefined?

jobara commented on extension/src/lib/chromeSettings.js:
```diff
@@ -0,0 +1,80 @@
+        oldSettings: null
+    },
+    invokers: {
+        onSettingsChange: {
```
This should probably be renamed. With the name starting with "on" it sounds more like an event. However, this could be used outside of the event system. Perhaps it should just be called "updateCurrentSettings" or something like that.

jobara commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,58 @@
+        getEnabled: {
+            funcName: "gpii.chrome.extensionHolder.getEnabled",
+            args: "{that}.extensionInstance"
+        }
```
Does the extensionInstance have any events for model changes that we could hook into. It seems strange to have get/set methods on the component that don't actually effect the component's model. Also perhaps these should be renamed to "setExtensionInstanceModel" or something like that. Also is it necessary to have the "get" invoker, as the value is easily accessible off of the `extensionInstance` member object.

javihernandez replied:
.



> Does the extensionInstance have any events for model changes that we could hook into.

No, it's static.

amb26 replied:
I bet it really does :) Noone would be insane enough to design an API with a volatile flag indicating its state without a means to observe it.

amb26 replied:
I win this bet: https://developer.chrome.com/extensions/management#event-onEnabled

javihernandez replied:
Yes, you can listen to events from extensions/apps (onEnabled, onInstalled, etc), but in this particular case I'm just performing a call to [get](https://developer.chrome.com/extensions/management#method-get), which returns an [ExtensionInfo](https://developer.chrome.com/extensions/management#type-ExtensionInfo) object (what I call extensionInstance, bad naming btw ...), but you can't listen to events from it. We can listen to events in general, and we can make some filtering to listen for events from a particular extension/app. It can probably be helpful for testing, but this is something I still need to deep dive into.

jobara commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+            fluid.each(tabs, function (tab) {
+                chrome.tabs.executeScript(tab.id, {code: that.options.disableScript}, function() {
+                    if (chrome.runtime.lastError) {
+                        console.log("Could not remove highContrast from tab '" +
```
Maybe covert the log messages to use [`fluid.log`](http://docs.fluidproject.org/infusion/development/CoreAPI.html#fluid-log-loglevel-arg1-argn-)

javihernandez replied:
> Maybe covert the log messages to use fluid.log

Yeah, but I'm not getting any log at all. Do I need to bundle anything else from infusion to get it working?

amb26 replied:
No, you just need to read the docs : P

http://docs.fluidproject.org/infusion/development/CoreAPI.html#logging-and-error-handling

amb26 replied:
use fluid.log

jobara commented on extension/src/lib/wsConnector.js:
```diff
@@ -0,0 +1,123 @@
+
+// Useful for testing
+gpii.wsConnector.onSettingsChange = function (settings) {
+    console.log("## Got the folowing settings: " + JSON.stringify(settings, null, 2));
```
This should probably also use fluid.log

jobara commented on extension/src/lib/zoom.js:
```diff
@@ -0,0 +1,50 @@
+    // Iterate over all tabs and set the zoom factor
+    chrome.tabs.query({}, function (tabs) {
+        fluid.each(tabs, function (tab) {
+            chrome.tabs.setZoom(tab.id, value, function () {
```
What happens if a user opens a new tab, after this is run?

javihernandez replied:
> What happens if a user opens a new tab, after this is run?

Fixed in highContrast and zoom components.

jobara commented on package.json:
```diff
@@ -0,0 +1,36 @@
+{
```
This seems to be missing the license information.

javihernandez replied:
added

jobara commented on tests/highContrastTests.js:
```diff
@@ -0,0 +1,33 @@
+
+    jqUnit.assertFalse("highContrastEnabled must be false at the beginning",
+                        highContrast.model.highContrastEnabled);
+    jqUnit.assertEquals("highContrastTheme must be undefined at the beginning",
```
You can use [`jqUnit.assertUndefined`](https://github.com/fluid-project/infusion/blob/master/tests/test-core/jqUnit/js/jqUnit.js#L150)

jobara commented on tests/highContrastTests.js:
```diff
@@ -0,0 +1,33 @@
+                        highContrast.model.highContrastTheme);
+
+    highContrast.applier.change("highContrastEnabled", true);
+    jqUnit.assertTrue("highContrastEnabled == true", highContrast.model.highContrastEnabled);
```
===

jobara commented on tests/highContrastTests.js:
```diff
@@ -0,0 +1,33 @@
+    highContrast.applier.change("highContrastTheme", "black-white");
+    jqUnit.assertEquals("highContrastTheme is set to black-white",
+                        "black-white",
+                        highContrast.model.highContrastTheme);
```
I guess we should also ensure that the model transformation is working.

jobara commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+});
+
+gpii.chrome.highContrast.modelChanged = function (that) {
+    if (that.model.highContrastEnabled) {
```
I wonder if these can be refactored into public functions and bound as model listeners to changes at specific model paths?

javihernandez replied:
The thing is that I need to evaluate both settings at the same time, doesn't matter which/when they change. That's the reason I'm using this approach, do you think there's an alternative?

amb26 replied:
In this case you want to listen on the path ["highContrastEnabled", "highContrastTheme"] using the longer modelListeners form: http://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#matching-on-multiple-paths-in-a-single-model-listener-declaration

jobara commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+
+gpii.chrome.highContrast.modelChanged = function (that) {
+    if (that.model.highContrastEnabled) {
+        var theme = fluid.model.transformWithRules(that.model, that.options.commonToPlatform);
```
This could be done with a modelRelay.

javihernandez replied:
Well, I've tried it but didn't work. :/ 
Is there any problem/restriction in using a valueMapper transformation inside a modelRelay?

amb26 replied:
No problem

javihernandez replied:
Ok, I wrote [this modelRelay](http://pastebin.com/STnF9sSH), but it isn't working. What am I doing wrong? It's always taking the defaultInputValue, why?

amb26 replied:
Looks like you've attached it to the wrong path - aren't the settings inside path "currentSettings"?

javihernandez replied:
Nope, defaultSettings is a member now and all the settings end up in the model itself. see: https://github.com/GPII/gpii-chrome-extension/pull/1/files#diff-a7885c5d019b65f034ed9db28a428799R55

javihernandez replied:
@jobara, @amb26 

is there anything wrong with [this modelRelay example](http://pastebin.com/rcCPdxEK)? I'm doing exactly the same here and it doesn't work.

jobara replied:
What error or result are you getting from that model relay?

javihernandez replied:
> What error or result are you getting from that model relay?

If I trigger a change on "numb" (setting it to "two"), child's "childNumb" is always "ONE" (the defaultInputValue). If I do not include a defaultInputValue it never gets transformed (childNumb is always null)

jobara replied:
I did a quick test of this with an Infusion component in the browser and things were working okay. How are you triggering the change on "numb"?

javihernandez replied:
> How are you triggering the change on "numb"?

parent.applier.change("numb", "two")

jobara replied:
I rewrote your child component as:

```javascript
            fluid.defaults("test.childComponent", {
                gradeNames: "fluid.modelComponent",
                model: {
                    childNumb: null
                },
                modelListeners: {
                    "childNumb": {
                    func: function (changeVal) {
                        console.log("childNumb:", changeVal);
                    },
                    args: ["{change}.value"]
                  }
                }
            });
```

and made the changes with an invoker:

```javascript
                invokers: {
                    update: {
                    changePath: "numb",
                    value: "{arguments}.0"
                  }
                },
```

jobara commented on extension/src/lib/wsConnector.js:
```diff
@@ -0,0 +1,123 @@
+            funcName: "gpii.wsConnector.onclose",
+            args: ["{that}",  "{arguments}.0"]
+        },
+        onmessage: {
```
As mentioned before, consider renaming these "on" invokers.

jobara commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,58 @@
+
+gpii.chrome.extensionHolder.populate = function (that) {
+    chrome.management.get(that.options.extensionId, function (extInfo) {
+        that.extensionInstance = extInfo;
```
consider making the callback function an invoker.

jobara commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,58 @@
+
+gpii.chrome.extensionHolder.setEnabled = function (that) {
+    that.extensionInstance.enabled = that.model.extensionEnabled;
+    chrome.management.setEnabled(that.extensionInstance.id, that.model.extensionEnabled, function () {
```
perhaps this callback could be an event.

javihernandez commented:
Hey @jobara,

I've just pushed some changes, although it's not ready yet, I appreciate any preliminary feedback.

amb26 commented on extension/src/lib/zoom.js:
```diff
@@ -0,0 +1,50 @@
+        }
+    },
+    modelListeners: {
+        "*": {
```
Should be "" rather than "*" - please see docs: http://docs.fluidproject.org/infusion/development/ChangeApplierAPI.html#wildcards-in-model-path-references

amb26 replied:
Also, if you use "" you can get rid of the "{that}" argument and just accept the default which will then just consist of the whole model - also change above. But in this case it looks like you only really want to listen to changes to "magnifierEnabled"

javihernandez replied:
Nope, I also want to listen to changes on "magnification". This is mainly to take care of updates in the current personalization triggered by a context change. i.e.: 
* I'm already logged in and I'm already using magnification
* At 10 pm I usually need a little bit more of magnification, so I've got a context that describes these preferences.

amb26 replied:
Either use "" or explicitly list the paths you wish to listen on, as per comment above

amb26 commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+
+        chrome.tabs.query({}, function (tabs) {
+            fluid.each(tabs, function (tab) {
+                chrome.tabs.executeScript(tab.id, {code: script}, function() {
```
Factor out this script execution and error handling logic into a self-contained function or component

javihernandez replied:
done

amb26 commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+        highContrastTheme: undefined
+    },
+    modelListeners: {
+        "*": {
```
See below

amb26 commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,87 @@
+//
+fluid.defaults("gpii.chrome.highContrast", {
+    gradeNames: ["fluid.modelComponent"],
+    scriptTemplate: "document.documentElement.setAttribute('data-gpii-hc','__THEME__');" +
```
Crikey, what is this! Don't we believe in putting scripts into files any more? : P

amb26 replied:
https://developer.chrome.com/extensions/content_scripts

javihernandez replied:
Ok, here's the thing.
I can inject this code as functions inside a content script (in fact I did it on the past), but it will be injected always, which is something I'd prefer not to do. Also, if the extension is enabled after the user has started using the browser, the content script won't be loaded into every window/tab, so we have to go over the tabs manually and make sure the content script is injected.
These are the reasons why I'm performing a [programmatic code injection](https://developer.chrome.com/extensions/content_scripts#pi). I've put the disableScript inside a file and injecting it when necessary, but the scriptTemplate needs some adjustments prior to be injected (replace __THEME__ with the current highcontrast theme). Besides storing the scriptTemplate as a file and reading it as a string .... do you think of any other ideas/alternatives?

javihernandez replied:
also, see: https://bugs.chromium.org/p/chromium/issues/detail?id=36400

amb26 replied:
Yes, please place these in files. As well as this, please return to the use of function calls with arguments (an accepted means of parameterising functionality since 1951) rather than template string substitution within code.

amb26 commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,112 @@
+//
+fluid.defaults("gpii.chrome.highContrast", {
+    gradeNames: ["fluid.modelComponent", "gpii.chrome.eventedComponent"],
+    scriptTemplate: "document.documentElement.setAttribute('data-gpii-hc','__THEME__');" +
```
See comment on old revision which has now been displaced

amb26 commented on extension/src/lib/wsConnector.js:
```diff
@@ -0,0 +1,115 @@
+            funcName: "gpii.wsConnector.setup",
+            args: "{that}"
+        },
+        onErrors: {
```
onError is a better name

amb26 commented on extension/src/lib/wsConnector.js:
```diff
@@ -0,0 +1,115 @@
+    }
+});
+
+gpii.wsConnector.onopen = function (that) {
```
All these invokers are unnecessary - they can just be written out inline in the "listeners" block

amb26 commented on extension/src/lib/zoom.js:
```diff
@@ -0,0 +1,71 @@
+var gpii = fluid.registerNamespace("gpii");
+var chrome = chrome || require("sinon-chrome");
+
+fluid.defaults("gpii.chrome.zoom", {
```
Rename to "chrome.thoom"

amb26 commented on extension/src/lib/zoom.js:
```diff
@@ -0,0 +1,71 @@
+gpii.chrome.zoom.applyZoomInTab = function (that, tab, value) {
+    chrome.tabs.setZoom(tab.id, value, function () {
+        if (chrome.runtime.lastError) {
+            console.log("Could not apply zoom in tab '" +
```
Use fluid.log :)

jobara commented on extension/src/lib/chromeSettings.js:
```diff
@@ -0,0 +1,73 @@
+        that.applier.change("", that.options.defaultSettings);
+    } else {
+        that.applier.change("", settings);
+    }
```
To answer your questions from the outdated diff ( https://github.com/GPII/gpii-chrome-extension/pull/1#discussion-diff-64906222 ) I don't think you could do the undefined check completely declaratively. @amb26 would know for sure.

If you do keep this function, you could simplify it by writing it like:

```javascript
gpii.chrome.settings.updateSettings = function (that, settings) {
    that.applier.change("", settings || that.options.defaultSettings);
};
```

jobara commented on Gruntfile.js:
```diff
@@ -0,0 +1,116 @@
+/* global module */
```
You could add a copyright to this file as well.

jobara commented on README.md:
```diff
@@ -1,2 +1,39 @@
+4. Navigate to the directory in which your local copy of the extension lives, and select the *build* folder.
+
+
+If you want to create a *crx* package to be distributed, run:
```
is the "crx" file for publishing the extension? Perhaps some more info on what this is and some links to docs would be useful.

jobara commented on README.md:
```diff
@@ -1,2 +1,39 @@
+
+    grunt buildPkg
+
+**Note that you need to use a .pem file to sign the crx package with. This file needs to be called *key.pem* and needs to be placed into the top level folder of this repository.**
```
Are there docs related to the ".pem" file that you could link to?

jobara commented on extension/src/lib/chromeEvented.js:
```diff
@@ -0,0 +1,31 @@
+fluid.defaults("gpii.chrome.eventedComponent", {
+    gradeNames: "fluid.component",
+    events: {
+        onCreate: null,
```
the onCreate event should already be added by the parent grade.

jobara commented on extension/src/lib/chromeEvented.js:
```diff
@@ -0,0 +1,31 @@
+        that.events.onTabOpened.fire(tab);
+    });
+    chrome.tabs.onUpdated.addListener(function (tabId, changeInfo, oldTabStatus) {
+        that.events.onTabUpdated.fire(oldTabStatus);
```
You may just want to pass along all of the arguments from the chrome event. The listener to infusion event can pick and choose what arguments it wants anyways, but this will make it more flexible.

jobara commented on extension/src/lib/zoom.js:
```diff
@@ -0,0 +1,72 @@
+gpii.chrome.zoom.applyZoomInTab = function (that, tab, value) {
+    chrome.tabs.setZoom(tab.id, value, function () {
+        if (chrome.runtime.lastError) {
+            fluid.log("Could not apply zoom in tab'",
```
maybe this should fire an event for the error.

jobara commented:
@javihernandez I tried the build and unpacked extension installation, which seemed to work well. I think the README is missing information about how the plugin actually works and what other requirements it has (e.g. GPII).

jobara commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,82 @@
+        }
+    },
+    listeners: {
+        onCreate: {
```
It's good practice to namespace listeners. For example this one could change to "onCreate.populate". Namespacing allows you to override an event in another grade, and also provides more control for ordering events.

http://docs.fluidproject.org/infusion/development/InfusionEventSystem.html#namespaced-listeners

javihernandez commented:
> @javihernandez I tried the build and unpacked extension installation, which seemed to work well. I think the README is missing information about how the plugin actually works and what other requirements it has (e.g. GPII).

Sure, will add some details, but it won't work until I create a pull request for universal that includes support for this solution. Meanwhile, I can provide you with details on how to test it if you want.

jobara commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,82 @@
+
+gpii.chrome.extensionHolder.setup = function (that, extInfo) {
+    if (chrome.runtime.lastError) {
+        fluid.log("Could not get extensionInfo, error was:",
```
You may want to consider using the fluid.logLevel option for the various logs you are emitting. 

http://docs.fluidproject.org/infusion/development/CoreAPI.html#fluid-loglevel

jobara commented on extension/src/lib/highContrast.js:
```diff
@@ -0,0 +1,120 @@
+    });
+};
+
+gpii.chrome.highContrast.formatScript = function (that, options) {
```
Is the `options` argument necessary. You don't seem to be using it when the function is assigned to the invoker.

javihernandez commented:
@jobara @amb26,

I've addressed more comments and also added more tests for both zoom and wsConnector. Is there any reason why the tests don't pass if I run all them at the same time from [all-tests.js](https://github.com/javihernandez/gpii-chrome-extension/blob/20db4f9bd0495d5b5727d0bcaf8b35e9069ca70f/tests/all-tests.js)?

javihernandez commented:
> Is there any reason why the tests don't pass if I run all them at the same time from all-tests.js?

Fixed in https://github.com/GPII/gpii-chrome-extension/pull/1/commits/ee523a05ac57af6f673a682433cef22310808626

jobara commented on extension/src/lib/chromeNotifications.js:
```diff
@@ -0,0 +1,117 @@
+
+gpii.chrome.notifications.create = function (that, options, eventFirer) {
+    // notificationId format is 'gpii-<milliseconds since 1st Jan. 1970>'
+    var notificationId = "gpii-".concat(new Date().getTime());
```
you could also use [fluid.allocateGuid](https://github.com/fluid-project/infusion/blob/master/src/framework/core/js/Fluid.js#L1145-L1150)

amb26 commented:
Some problems at top level:
i) linting task fails
ii) Because linting task fails, buildPkg grunt task can't run
iii) Disabling linting task yields the following failure on crx:build subtask of grunt buildPkg:
Running "crx:build" (crx) task
Warning: Cannot find module 'build\manifest.json' Use --force to continue.

amb26 commented on .eslintrc.json:
```diff
@@ -0,0 +1,79 @@
+{
```
As you fix the linting issues, could you please update this to our latest linting style using the project-wide shared configuration? Example is at https://github.com/fluid-project/kettle/blob/master/.eslintrc.json

javihernandez replied:
done

amb26 commented on README.md:
```diff
@@ -1,2 +1,67 @@
+### Testing the auto-personalization
+
+We have included two users in the GPII in order to demonstrate this extension:
+* *gpiiChrome1*: Enables chromeVox
```
I couldn't find these preference sets in trunk or any pull request - is there a matching universal pull request which needs to be issued?

javihernandez replied:
Yes, this needs to be issued, will try to get it ready today.

javihernandez replied:
Ok, now you can use _chrome1_ and _chrome2_ from https://github.com/GPII/universal/pull/500

amb26 commented on extension/src/lib/chromeEvented.js:
```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+var gpii = fluid.registerNamespace("gpii");
+var chrome = chrome || require("sinon-chrome");
```
Repeating this global boilerplate mentioning a particular test library is a bit awkward. Add a TODO here to make use of the Infusion ContextAwareness API - http://docs.fluidproject.org/infusion/development/ContextAwareness.html - no need to do this for this pull request

javihernandez replied:
done

amb26 commented on extension/src/lib/chromeNotifications.js:
```diff
@@ -0,0 +1,117 @@
+    });
+};
+
+gpii.chrome.notifications.update = function (that, notificationId, options, eventFirer) {
```
Factor away all these repeated functions into a single handler

javihernandez replied:
@amb26, not all of them share the same signature (https://developer.chrome.com/apps/notifications) and also, we need to have access to both _that_ and _eventFirer_ arguments during the callback function. I have wrapped the if-else code into a function _handleResult_ but not sure how I can make this even more compact in a declarative way. Any advice?

In any case, some of these api calls are not being used at this moment (actually we only make use of _create_ and _clear_). I just put them there just in case. I guess that we can just get rid of them if you don't think it's worth refactoring this and/or if we want to keep this simple.

amb26 commented on extension/src/lib/chromeSettings.js:
```diff
@@ -0,0 +1,143 @@
+            type: "basic",
+            title: "GPII notifications",
+            message: extension.options.name + " couldn't be found. Do you want to add it from the chrome store?",
+            iconUrl: "https://raw.githubusercontent.com/javihernandez/android/master/platform/app/res/drawable-hdpi/gpii_logo.png",
```
This is a pretty funny URL! Perhaps it would be better to source this from within the extension itself? I assume there is a scheme for bundling extension resources.

javihernandez replied:
done

amb26 commented on extension/src/lib/extensionHolder.js:
```diff
@@ -0,0 +1,92 @@
+    chrome.management.get(that.options.extensionId, that.setup);
+};
+
+gpii.chrome.extensionHolder.switchStatus = function (that) {
```
Rename to updateEnabledStatus

amb26 commented on README.md:
```diff
@@ -1,2 +1,67 @@
+
+You can also run the browser tests, open your browser and go to:
+
+    chrome-extension://ifommhafkphfognacgcmpidnpbffedlp/tests/highContrastTests.html
```
I tried this URL and got "This site can’t be reached"

javihernandez replied:
Well, probably I have to update the README file to indicate that, since the extension id will be different, you need to copy/paste it from the extension's section of chrome.

javihernandez replied:
done

javihernandez commented:
@amb26 this should be ready for a new review.
